zeebe/engine/src/main/java/io/camunda/zeebe/engine/state/conditional/DbConditionalSubscriptionState.java

```diff
@@ -117,6 +129,35 @@ public void deleteStart(final long key, final ConditionalSubscriptionRecord subs
         processDefinitionKeyAndTenantAwareSubscriptionKey);
   }
 
+  private void incrementProcessDefinitionKeySubscriptionCount(
+      final ConditionalSubscriptionRecord subscription) {
+    processDefinitionKey.wrapLong(subscription.getProcessDefinitionKey());
+    final DbInt dbInt = processDefinitionKeyCountColumnFamily.get(processDefinitionKey);
+    if (dbInt != null) {
+      subscriptionCount.wrapInt(dbInt.getValue() + 1);
+    } else {
+      final DbInt newCount = new DbInt();
+      newCount.wrapInt(1);
+      subscriptionCount.wrapInt(newCount.getValue());
+    }
+    processDefinitionKeyCountColumnFamily.upsert(processDefinitionKey, subscriptionCount);
+  }
+
+  private void decrementProcessDefinitionKeySubscriptionCount(
+      final ConditionalSubscriptionRecord subscription) {
+    processDefinitionKey.wrapLong(subscription.getProcessDefinitionKey());
+    final DbInt dbInt = processDefinitionKeyCountColumnFamily.get(processDefinitionKey);
+    if (dbInt != null) {
+      final int newCount = dbInt.getValue() - 1;
+      if (newCount == 0) {
+        processDefinitionKeyCountColumnFamily.deleteExisting(processDefinitionKey);
```

Comment:
🔧 While unexpected, an underflow would accidentally lead to keeping the entry in the column family, breaking the usage of the exist method to check for no subscriptions. I suggest checking for `<= 0` or dealing with the case explicitly (logging a warning or even considering it an inconsistency and throwing an exception)